packages/payload/src/fields/hooks/beforeChange/promise.ts

```diff
@@ -98,6 +94,11 @@ export const promise = async ({
   const schemaPathSegments = schemaPath ? schemaPath.split('.') : []
   const indexPathSegments = indexPath ? indexPath.split('-').filter(Boolean)?.map(Number) : []
 
+  const passesCondition = field.admin?.condition
+    ? Boolean(field.admin.condition(data, siblingData, { blockData, user: req.user }, pathSegments))
```

Comment:
Looks like the `ctx` object does not only contain `user`, but also `blockData`. We need to put the `path` here as well instead of passing a fourth argument.